@here/harp-geoutils/lib/projection/Projection.ts

```diff
@@ -123,43 +123,93 @@ export interface Projection {
+    abstract unprojectBox(worldBox: Box3Like): GeoBox;
 
     /**
      * Returns the scaling factor that must be used to convert the units used by `worldPoint` to
      * meters.
      *
      * @param worldPoint The position in world coordinates.
      */
-    getScaleFactor(worldPoint: Vector3Like): number;
+    abstract getScaleFactor(worldPoint: Vector3Like): number;
 
     /**
      * Returns the surface normal at the given world position.
      *
      * @param worldPoint The position in world coordinates.
      */
-    surfaceNormal(worldPoint: Vector3Like): Vector3Like;
+    abstract surfaceNormal(worldPoint: Vector3Like): Vector3Like;
 
     /**
      * Returns the surface normal at the given world position.
      *
      * @param worldPoint The position in world coordinates.
      * @returns The resulting normal vector.
      */
-    surfaceNormal<Normal extends Vector3Like>(worldPoint: Vector3Like, result: Normal): Normal;
+    abstract surfaceNormal<Normal extends Vector3Like>(
+        worldPoint: Vector3Like,
+        result: Normal
+    ): Normal;
 
     /**
      * Returns the signed distance between the given coordinates and
      * the closest point on the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    groundDistance(worldPoint: Vector3Like): number;
+    abstract groundDistance(worldPoint: Vector3Like): number;
 
     /**
      * Scales the given world coordinates to the surface.
      *
      * @param worldPoint The position in world coordinates.
      */
-    scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+    abstract scalePointToSurface(worldPoint: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid world position for the given source projection.
+     * @returns The world position reprojected using this [[Projection]].
+     */
+    reprojectPoint(sourceProjection: Projection, worldPos: Vector3Like): Vector3Like;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     */
+    reprojectPoint<WorldCoordinates extends Vector3Like>(
+        sourceProjection: Projection,
+        worldPos: Vector3Like,
+        result: WorldCoordinates
+    ): typeof result;
+
+    /**
+     * Reproject a world position from the given source [[Projection]].
+     *
+     * @param sourceProjection The source projection.
+     * @param worldPos A valid position in the world space defined by the source projection.
+     * @param result The resulting position reprojected using this [[Projection]].
+     * @hidden
+     */
+    reprojectPoint(
```

Comment:
Why do we need to declare the method three times? Is this implementation not enough?

Reply:
it's for avoiding possible runtime errors when using the `Projection` APIs. In fact, without restricting the signatures the following code will result in a runtime error:
```
const p = projection.reprojectPoint<THREE.Vector3>(sourceProjection, worldP);
p.set(0,0,0); // runtime error THREE.Vector3
```

On the other hand, the restricted APIs works as expected:
```
const p = projection.reprojectPoint(sourceProjection, worldP);
const {x,y,z} = p; // OK, p is a Vector3Like
p.sub(0, 0, 0); // OK, type error, Vector3Like doesn't have methods.

const p = projection.reprojectPoint(sourceProjection, worldP, new THREE.Vector3());
p.set(0, 0, 0); // OK, p is a THREE.Vector3.
const {x,y,z} = p; // OK, THREE.Vector3 is compatible with Vector3Like
```

Reply:
Thanks for the detailed explanation.